Managers/FrameMan.cpp

```diff
@@ -1,2651 +1,1391 @@
+						}
+					} else {
+						if (m_TextCentered[whichScreen]) { yTextPos = (GetPlayerScreenHeight() / 2) - 52; }
+
+						int occOffsetX = g_SceneMan.GetScreenOcclusion(whichScreen).m_X;
+
+						// If there's really no room to offset the text into, then don't
+						if (GetPlayerScreenWidth() <= GetResX() / 2)
+							occOffsetX = 0;
+						// Draw blinking effect, but not of the text message itself, but some characters around it (so it's easier to read the message)
+						if (m_TextBlinking[whichScreen] && m_TextBlinkTimer.AlternateReal(m_TextBlinking[whichScreen])) {
+							GetLargeFont()->DrawAligned(&pPlayerGUIBitmap, (GetPlayerScreenWidth() + occOffsetX) / 2, yTextPos, (">>> " + m_ScreenText[whichScreen] + " <<<").c_str(), GUIFont::Centre);
+						} else {
+							GetLargeFont()->DrawAligned(&pPlayerGUIBitmap, (GetPlayerScreenWidth() + occOffsetX) / 2, yTextPos, m_ScreenText[whichScreen].c_str(), GUIFont::Centre);
+						}
+					}
+					yTextPos += 12;
+				}
+				g_PerformanceMan.Draw(pPlayerGUIBitmap);
+
+				// Draw info text when in MOID or material layer draw mode
+				switch (g_SceneMan.GetLayerDrawMode()) {
+					case g_LayerTerrainMatter:
+						GetSmallFont()->DrawAligned(&pPlayerGUIBitmap, GetPlayerScreenWidth() / 2, GetPlayerScreenHeight() - 12, "Viewing terrain material layer\nHit Ctrl+M to cycle modes", GUIFont::Centre, GUIFont::Bottom);
+						break;
+					case g_LayerMOID:
+						GetSmallFont()->DrawAligned(&pPlayerGUIBitmap, GetPlayerScreenWidth() / 2, GetPlayerScreenHeight() - 12, "Viewing MovableObject ID layer\nHit Ctrl+M to cycle modes", GUIFont::Centre, GUIFont::Bottom);
+						break;
+					default:
+						break;
+				}
 
+				// If superfluous screen (as in a three-player match), make the fourth the Observer one
+			} else {
+				//yTextPos += 12;
+				GetLargeFont()->DrawAligned(&pPlayerGUIBitmap, GetPlayerScreenWidth() / 2, yTextPos, "- Observer View -", GUIFont::Centre);
+			}
 
-//////////////////////////////////////////////////////////////////////////////////////////
-// Method:          DrawBitmapPrimitive
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Schedule to draw a bitmap primitive.
-// Arguments:       Player screen to draw primitive on. Position of primitive in scene coordintaes, an entity to sraw sprite from, 
-//					rotation angle in radians, frame to draw
-// Return value:    None.
-
-	void FrameMan::DrawBitmapPrimitive(int player, Vector start, Entity * pEntity, float rotAngle, int frame)
-	{
-		MOSprite * pMOS = dynamic_cast<MOSprite *>(pEntity);
-		if (pMOS)
-		{
-			BITMAP * pBitmap = pMOS->GetSpriteFrame(frame);
-			if (pBitmap)
-				m_Primitives.push_back(new BitmapPrimitive(player, start, pBitmap, rotAngle));
+			// If we are dealing with split screens, then deal with the fact that we need to draw the player screens to different locations on the final buffer
```

Comment:
This comment ("If we are dealing with split screens...") should be below since it applies to that stuff not the Vector screenOffset line. It being where it is is also confusing cause it makes it look like they're one comment that got split over two lines.

Also, pull the splitscreen handling here out into a separate private method. In order to modify screenOffset properly you'll need to take in a reference to screenOffset, i.e.
```
FrameMan::UpdateScreenOffsetForSplitScreen(Vector &screenOffset) {...}
```
You'd call it with
```
if (screenCount > 1) { UpdateScreenOffsetForSplitScreen(screenOffset); }
```